### PR TITLE
[Feature]: Support loss inputs routing

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR types

New features
### PR changes

APIs

### CODE REVIEW

File: paddleseg/core/train.py [hunk 1]
```diff
@@ -25,34 +25,19 @@
...
         if loss_i.__class__.__name__ in ('BCELoss', ) and loss_i.edge_label:
             # Use edges as labels According to loss type.
-            loss_list.append(coef_i * loss_i(logits, edges))
-        elif loss_i.__class__.__name__ == 'MixedLoss':
-            mixed_loss_list = loss_i(logits, labels)
-            for mixed_loss in mixed_loss_list:
-                loss_list.append(coef_i * mixed_loss)
-        elif loss_i.__class__.__name__ in ("KLLoss", ):
-            loss_list.append(coef_i *
-                             loss_i(logits_list[0], logits_list[1].detach()))
```
Comment (Asthestarsfalll) [inline paddleseg/core/train.py:53]: 可以将两个判断都消去，对于KLLoss，需要将detach移入froward中，Mixed Loss则可以直接删除。希望将edge的判断也消除，或许可以考虑直接传入data字典？

File: paddleseg/cvlibs/builder.py [hunk 2]
```diff
@@ -235,6 +235,24 @@ def _check_helper(loss_cfg, ignore_index):
...
+        def _check_output_index(output_index):
+            all_index = []
+            for item in output_index:
+                all_index.extend(item)
+            all_index.sort()
+            all_index = list(set(all_index))
+            target = [i for i in range(max(all_index))]
+            if all_index != target:
+                raise ValueError(
+                    'The values of `output_index` must be continuous intergers')
```
Comment (Asthestarsfalll) [inline paddleseg/cvlibs/builder.py:254]: 检查了一下是否覆盖了0-n的连续整数，确保是否使用了所有输出，但是不知道具体有多少输出

File: paddleseg/cvlibs/config_checker.py [hunk 3]
```diff
@@ -88,6 +88,13 @@ def check_and_correct(self, cfg):
                 'Loss config should contain keys of "types" and "coef"'
         len_types = len(loss_cfg['types'])
         len_coef = len(loss_cfg['coef'])
+        repeat = loss_cfg.get('repeat', None)
+
+        if repeat:
+            loss_cfg['types'] = loss_cfg['types'] * repeat
+            if len_types == len_coef:
+                loss_cfg['coef'] = loss_cfg['coef'] * repeat
```
Comment (Asthestarsfalll) [inline paddleseg/cvlibs/config_checker.py:96]: 由于去除了Mixed Loss，不符合后续自动扩展`len_types == 1`的条件，这里用repeat省事一点。第二个判断适用于coef都为一个值的情况，若不然，则需要全部手写

File: paddleseg/cvlibs/builder.py [hunk 4]
```diff
@@ -235,6 +235,24 @@ def _check_helper(loss_cfg, ignore_index):
                     'the ignore_index in loss and train_dataset must be the same. Currently, loss ignore_index = {}, '\
                     'train_dataset ignore_index = {}'.format(loss_cfg['ignore_index'], ignore_index)
 
+        def _get_default_output_index(loss_cfg):
+            length = len(loss_cfg['types'])
+            repeat = loss_cfg.get('repeat', None)
+            item_len = length // repeat if repeat else 1
+            length *= item_len
+            return [[i // item_len] for i in range(length)]
```
Comment (Asthestarsfalll) [inline paddleseg/cvlibs/builder.py:243]: 向后兼容，没有设置时`output_index`默认为`[[0], [1] ... [n]]`，当有repeat参数时，如repeat=2, 为`[[0], [0], [1], [1] .... [n], [n]]`，因此当未指定`output_index`时， 使用repeat表示mixed loss
